### PR TITLE
item components

### PR DESCRIPTION
**Pull request by eristhea**
based on #557, as I needed that work to make sense of a lot of this
names lean towards Mojang's names a lot more than Yarn's, as I find them more intuitive

### CODE REVIEW

File: mappings/net/minecraft/client/util/ColorUtil.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/util/ColorUtil.mapping:1]: I think `ARGB32` should be `Argb32`

File: mappings/net/minecraft/client/util/ColorUtil.mapping [hunk 1]
```diff
@@ -15,6 +24,8 @@ CLASS net/minecraft/unmapped/C_tmbijhir net/minecraft/client/util/ColorUtil
 			ARG 0 argb
 		METHOD m_tpoxzqsm getAlpha (I)I
 			ARG 0 argb
+		METHOD m_xrcgvmaq opaque (I)I
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/util/ColorUtil.mapping]:
```suggestion
		METHOD m_xrcgvmaq toOpaque (I)I
```

File: mappings/net/minecraft/component/type/ItemEnchantmentsComponent.mapping [hunk 2]
```diff
@@ -0,0 +1,43 @@
+CLASS net/minecraft/unmapped/C_qlclfxvn net/minecraft/component/type/ItemEnchantmentsComponent
+	FIELD f_nekwvric FULL_CODEC Lcom/mojang/serialization/Codec;
+	FIELD f_osezxuts DEFAULT Lnet/minecraft/unmapped/C_qlclfxvn;
+	FIELD f_svrotkyo LEVEL_CODEC Lcom/mojang/serialization/Codec;
+	FIELD f_weawpons LEVELS_CODEC Lcom/mojang/serialization/Codec;
+	FIELD f_zenuxhpu shownInTooltip Z
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/component/type/ItemEnchantmentsComponent.mapping:6]:
```suggestion
	FIELD f_zenuxhpu showInTooltip Z
```

File: mappings/net/minecraft/component/type/PotionContentsComponent.mapping [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	METHOD m_linsoxal getColor (Ljava/lang/Iterable;)I
+	METHOD m_lwwcxstm getColor (Lnet/minecraft/unmapped/C_cjzoxshv;)I
+		ARG 0 potion
+	METHOD m_mgymqmlt customEffects ()Ljava/util/List;
+	METHOD m_nghpasjm (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/kinds/App;
+		ARG 0 instance
+	METHOD m_njngtrku matches (Lnet/minecraft/unmapped/C_cjzoxshv;)Z
+		ARG 1 potion
+	METHOD m_nlaiylfj getColor ()I
+	METHOD m_oixxcdng getColorOptional (Ljava/lang/Iterable;)I
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/component/type/PotionContentsComponent.mapping:26]:
```suggestion
	METHOD m_oixxcdng tryGetColor (Ljava/lang/Iterable;)I
```
?

File: mappings/net/minecraft/client/util/ColorUtil.mapping [hunk 4]
```diff
@@ -5,6 +5,15 @@ CLASS net/minecraft/unmapped/C_tmbijhir net/minecraft/client/util/ColorUtil
 		METHOD m_gjglijic mixColor (II)I
 			ARG 0 first
 			ARG 1 second
+		METHOD m_jdoowdec getArgb (III)I
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/util/ColorUtil.mapping]:
```suggestion
		METHOD m_jdoowdec of (III)I
```
or
```suggestion
		METHOD m_jdoowdec toArgb (III)I
```

File: mappings/net/minecraft/client/util/ColorUtil.mapping [hunk 5]
```diff
@@ -5,6 +5,15 @@ CLASS net/minecraft/unmapped/C_tmbijhir net/minecraft/client/util/ColorUtil
...
+			ARG 0 red
+			ARG 1 blue
+			ARG 2 green
+		METHOD m_jslmgbyf getArgb (IIII)I
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/util/ColorUtil.mapping]:
```suggestion
		METHOD m_jslmgbyf of (IIII)I
```
or
```suggestion
		METHOD m_jslmgbyf toArgb (IIII)I
```

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 6]
```diff
@@ -111,23 +128,43 @@ CLASS net/minecraft/unmapped/C_sddaxwyk net/minecraft/item/ItemStack
...
 	METHOD m_iculdxhv isDamageable ()Z
+	METHOD m_igidwirs getComponentPatch ()Lnet/minecraft/unmapped/C_jqyoylib;
 	METHOD m_ihhujymf toHoverableText ()Lnet/minecraft/unmapped/C_rdaqiwdt;
 	METHOD m_imaclehj useOnEntity (Lnet/minecraft/unmapped/C_jzrpycqo;Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_laxmzoqs;)Lnet/minecraft/unmapped/C_ozuepbyj;
 		ARG 1 user
 		ARG 2 entity
 		ARG 3 hand
+	METHOD m_imgzslzn listHashCode (Ljava/util/List;)I
+		ARG 0 stacks
+	METHOD m_izgziaft fromNbtOptional (Lnet/minecraft/unmapped/C_vtbxyypo$C_etmlgbig;Lnet/minecraft/unmapped/C_oivssbvb;)Ljava/util/Optional;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/item/ItemStack.mapping:149]:
```suggestion
	METHOD m_izgziaft tryFromNbt (Lnet/minecraft/unmapped/C_vtbxyypo$C_etmlgbig;Lnet/minecraft/unmapped/C_oivssbvb;)Ljava/util/Optional;
```
?

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 7]
```diff
@@ -173,6 +223,11 @@ CLASS net/minecraft/unmapped/C_sddaxwyk net/minecraft/item/ItemStack
 	METHOD m_sfmosozi getItemBarColor ()I
 		COMMENT {@return the color of the filled section of the durability bar}
 	METHOD m_sjztniec isStackable ()Z
+	METHOD m_skenhwqy stackListsMatch (Ljava/util/List;Ljava/util/List;)Z
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/item/ItemStack.mapping]:
```suggestion
	METHOD m_skenhwqy areEqual(Ljava/util/List;Ljava/util/List;)Z
```
or does this being an overload cause issues?

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 8]
```diff
@@ -214,6 +276,7 @@ CLASS net/minecraft/unmapped/C_sddaxwyk net/minecraft/item/ItemStack
 		ARG 3 remainingUseTicks
 	METHOD m_wnspogky streamTags ()Ljava/util/stream/Stream;
 		COMMENT {@return a stream of tag keys this item is in}
+	METHOD m_wxdafsoy checkCount (Lnet/minecraft/unmapped/C_sddaxwyk;)Lcom/mojang/serialization/DataResult;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/item/ItemStack.mapping]:
```suggestion
	METHOD m_wxdafsoy verifyCount (Lnet/minecraft/unmapped/C_sddaxwyk;)Lcom/mojang/serialization/DataResult;
```
?

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 9]
```diff
@@ -226,14 +289,22 @@ CLASS net/minecraft/unmapped/C_sddaxwyk net/minecraft/item/ItemStack
 		ARG 1 holder
 	METHOD m_xvzvtufo getName ()Lnet/minecraft/unmapped/C_rdaqiwdt;
 	METHOD m_yawcimjo getEntityHolder ()Lnet/minecraft/unmapped/C_astfners;
+	METHOD m_yqbabycc createOptionalCodec (Ljava/lang/String;)Lcom/mojang/serialization/MapCodec;
+		ARG 0 fieldName
+	METHOD m_yqsxbejh transmuteCopy (Lnet/minecraft/unmapped/C_gmbqjnle;I)Lnet/minecraft/unmapped/C_sddaxwyk;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/item/ItemStack.mapping]: transmute?

File: mappings/net/minecraft/item/ItemStack.mapping [hunk 10]
```diff
@@ -226,14 +289,22 @@ CLASS net/minecraft/unmapped/C_sddaxwyk net/minecraft/item/ItemStack
...
 		ARG 2 context
+	METHOD m_yzsowcuq (Lnet/minecraft/unmapped/C_cpwnhism;)Lnet/minecraft/unmapped/C_cpwnhism;
+		ARG 1 style
+	METHOD m_zakagwqa set (Lnet/minecraft/unmapped/C_pscqxfcs;Ljava/lang/Object;)Ljava/lang/Object;
 	METHOD m_zpjrlkav onCraftByPlayer (Lnet/minecraft/unmapped/C_cdctfzbn;Lnet/minecraft/unmapped/C_jzrpycqo;I)V
 		ARG 1 world
 		ARG 2 player
 		ARG 3 amount
 	METHOD m_zszkcovn hasEnchantments ()Z
+	METHOD m_zzjqqsgn transmuteCopyIgnoreEmpty (Lnet/minecraft/unmapped/C_gmbqjnle;I)Lnet/minecraft/unmapped/C_sddaxwyk;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/item/ItemStack.mapping]: transmute?

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/item/ItemStack.mapping (hunk 10 above)
Comment (eristhea) [reply mappings/net/minecraft/item/ItemStack.mapping]: I'll add javadocs to this method, it's pretty complicated
 there's definitely a better name

File: mappings/net/minecraft/component/type/ItemEnchantmentsComponent.mapping (hunk 2 above)
Comment (eristhea) [reply mappings/net/minecraft/component/type/ItemEnchantmentsComponent.mapping:6]: for fields my name follows convention better, that would be the method name

File: mappings/net/minecraft/component/type/ItemEnchantmentsComponent.mapping [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+	METHOD m_rqlwgwxc (Lnet/minecraft/unmapped/C_qlclfxvn;)Ljava/lang/Boolean;
+		ARG 0 component
+	METHOD m_szyctegw getEnchantmentEntries ()Ljava/util/Set;
+		COMMENT @return a map with enchantments as keys and their levels as values
+	METHOD m_ugcgsgdf getLevel (Lnet/minecraft/unmapped/C_jxtrubuh;)I
+		ARG 1 enchantment
+	METHOD m_ylenfqyd (Lnet/minecraft/unmapped/C_qlclfxvn;)Lit/unimi/dsi/fastutil/objects/Object2IntLinkedOpenHashMap;
+		ARG 0 component
+	CLASS C_vbojeisc Builder
+		FIELD f_lvvemeqz showInTooltip Z
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/component/type/ItemEnchantmentsComponent.mapping]:
```suggestion
		FIELD f_lvvemeqz shownInTooltip Z
```

File: mappings/net/minecraft/client/util/ColorUtil.mapping
Comment (eristhea) [reply mappings/net/minecraft/client/util/ColorUtil.mapping:1]: I agree, I'll do that refactor here

File: mappings/net/minecraft/component/type/PotionContentsComponent.mapping (hunk 3 above)
Comment (eristhea) [reply mappings/net/minecraft/component/type/PotionContentsComponent.mapping:26]: I prefer the `optional` suffix over the `try` prefix for methods like this

File: mappings/net/minecraft/item/ItemStack.mapping (hunk 10 above)
Comment (eristhea) [reply mappings/net/minecraft/item/ItemStack.mapping]: imagine my disappointment when I stare at these for 5 minutes and realise that the better name I've come up with is just yarn's

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/component/type/PotionContentsComponent.mapping (hunk 3 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/component/type/PotionContentsComponent.mapping:26]: Would you consider `getOptionalColor` (and `optionalFromNbt`)?
I think having the adjective follow the noun makes it more awkward to read.

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/component/type/PotionContentsComponent.mapping:26]: discussed on discord

**Review by moehreag — APPROVED** [review]

**Pull request merged**